packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -211,26 +206,26 @@ function mountCustomElement(vnode: VCustomElement, parent: ParentNode, anchor: N
         vm = createViewModelHook(elm, vnode);
     });
 
+    setScopeTokenClassIfNecessary(elm, owner);
+    if (owner.shadowMode === ShadowMode.Synthetic) {
+        const { stylesheetToken } = owner.context;
+        // when running in synthetic shadow mode, we need to set the shadowToken value
+        // into each element from the template, so they can be styled accordingly.
+        if (!isUndefined(stylesheetToken)) {
+            setElementShadowToken(elm, stylesheetToken);
+        }
+    }
+
     linkNodeToShadow(elm, owner);
     vnode.elm = elm;
     vnode.vm = vm;
 
     if (vm) {
         allocateChildren(vnode, vm);
-    } else if (vnode.ctor !== UpgradableConstructor) {
```

Comment:
this is another observable difference, because core doesn't care anymore, you might simply rely on a globally defined element while resolving to something completely unrelated. This makes is easier to eventually introduce the concept of external/global custom elements into templates. Eventually the ctor value can be `NULL`